File: Programmers/JavaScript/숫자 변환하기.js
const solution = (x, y, n) => {
    const dp = new Array(1_000_001).fill(Infinity);
    dp[x] = 0;

    for (let i = x; i < y + 1; i++){
        const a = i - n;
        const b = Number.isInteger(i / 2) ? i / 2 : -1;
        const c = Number.isInteger(i / 3) ? i / 3 : -1;

        [a,b,c].forEach(j => {
            if (j !== -1 && x <= j && dp[j] !== Infinity){
                dp[i] = Math.min(dp[i], dp[j] + 1);
            }
        })
    }
    const answer = dp[y] !== Infinity ? dp[y] : -1;

    return answer;
}

File: Programmers/숫자 변환하기.py
def solution(x, y, n):
    INF = 10_000_000
    dp = [INF] * 1_000_001
    dp[x] = 0

    for i in range(x, y + 1):
        a = i - n
        b = int(i / 3) if i % 3 == 0 else -1
        c = int(i / 2) if i % 2 == 0 else -1

        indexes = [a, b, c]

        for j in indexes:
            if j % 1 == 0 and x <= j and dp[j] != INF:
                dp[i] = min(dp[i], dp[j] + 1)

    answer = dp[y] if dp[y] != INF else -1

    return answer